crud/compare/conditions.lua

```diff
@@ -77,6 +78,33 @@ local funcs_by_symbols = {
     ['>='] = conditions.funcs.ge,
 }
 
+function conditions.get_index_for_condition(space_indexes, space_format, condition)
+    -- If we use # (not table.maxn), we may lose indexes, when user drop some indexes.
+    -- E.g: we have table with indexes id {1, 2, 3, nil, nil, 6}.
+    -- If we use #{1, 2, 3, nil, nil, 6} (== 3) we will lose index with id = 6.
+    -- See details: https://github.com/tarantool/crud/issues/103
+    local max_index = table.maxn(space_indexes)
```

Comment:
Please add a unit test that covers case with `nil` in `space_indexes`.

Reply:
Added check on higher level (in base_plan):

```lua
local space_indexes = space.index
if space_indexes == nil or not next(space_indexes) then
    return nil, NoIndexesError:new('Space has no indexes, space should have primary and bucket_id indexes')
end
```

And integration tests for `select`, `count`:

```lua
pgroup.test_select_no_index = function(g)
     local obj, err = g.cluster.main_server.net_box:call(
         'crud.select', {'no_index_space', nil}
     )

     t.assert_equals(obj, nil)
     t.assert_str_contains(err.err, "Space has no indexes, space should have primary and bucket_id indexes")
 end
```

```lua
pgroup.test_count_no_index = function(g)
     local obj, err = g.cluster.main_server.net_box:call(
         'crud.count, {'no_index_space', nil}
     )

     t.assert_equals(obj, nil)
     t.assert_str_contains(err.err, "Space has no indexes, space should have primary and bucket_id indexes")
 end
```

Reply:
Note: File `plan.lua` was moved to `compare` folder.